bluesky/callbacks/broker.py

```diff
@@ -7,7 +7,32 @@
 import matplotlib.pyplot as plt
 
 
-class LiveImage(CallbackBase):
+class BrokerCallbackBase(CallbackBase):
+    """
+    Base class for callbacks which need filled documents
+
+    Parameters
+    ----------
+    fields: Iterable of str
+        Names of data field in an Event
+    fs: FileStore instance
+        The FileStore instance to pull the data from
+    """
+    def __init__(self, fields, *, fs=None):
+        if fs is None:
+            import filestore.api as fs
```

Comment:
I think that it might be good to leave filestore optional. We only require it if the document is to be filled. The subclassed callbacks can have either pre-loaded data (coming from data analysis) or stored data (from the RunEngine) in the case of data analysis there isn't a concept of a filestore.

We may want to further separate this down the line, eg have a two node pipeline: the first loads the data in from wherever (if needed) and the second performs the callback. This way the callbacks are completely blind to the source of the data, it is always loaded by the time it gets to them.